pkg/db/sqlite.go

```diff
@@ -55,13 +72,46 @@ func (sc *SqliteConnection) Query(query string, args ...interface{}) (DataRows,
 
 func (sc *SqliteConnection) Exec(query string, args ...interface{}) (sql.Result, error) {
 	sc.logger.Debugf("Sqlite3 Exec(): %s | %v", query, args)
+	if !sc.validate(query) {
+		sc.logger.Errorf("Regex validation for query '%s' failed", query)
+		if !sc.executeUnverified {
+			return nil, fmt.Errorf("Regex validation for query '%s' failed", query)
+		}
+	}
 	result, err := sc.db.Exec(query, args...)
 	if err != nil {
 		sc.logger.Errorf("Sqlite3 Exec() error: %s", err)
 	}
 	return result, err
 }
 
+func (sc *SqliteConnection) validate(query string) bool {
```

Comment:
From SQL perspective, the differences between SQLite and Postgres are (except timestamps and custom DB-functions) shoulnd't be that extreme. Would it be possible to centralize the SQL validation logic in a SQL-validator object and pass that object to the DB-connections?

By making the validator configurable to enable/disable the validation internally, e.g.

    type sqlValidator struct {
        blockQueries bool
    }
    
    func newValidator(blockQueries bool) *sqlValidator {
        return &sqlValidator{blockQueries}
    }
    
     func (s *sqlValidator) Validate(sql string) error {
         //validte SQL query here...
         msg := fmt.Sprintf("Found potential SQL injection: %s", sql)
         if s.blockQueries {
            return errors.New(msg)
        } 
        log.Warnf(msg)
        return nil
     }